core/src/main/java/io/parsingdata/metal/data/ParseGraph.java

```diff
@@ -97,15 +100,15 @@ ParseGraph closeBranch() {
      * @return The first value (bottom-up) in this graph
      */
     public ParseValue current() {
-        if (isEmpty()) { return null; }
-        if (head.isValue()) {
-            return head.asValue();
-        }
-        if (head.isGraph()) {
-            final ParseValue value = head.asGraph().current();
-            if (value != null) { return value; }
-        }
-        return tail.current(); // Ignore current if it's a reference (or an empty graph)
```

Comment:
I removed it on purpose because it doesn't add much to the understanding of the code but could turn stale if we ever add another type of `ParseItem`.